remote/mattermost/remote.go

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package mattermost
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	"github.com/target/flottbot/models"
+	"github.com/target/flottbot/remote"
+)
+
+// Client struct.
+type Client struct {
+	Server   string
+	Token    string
+	BotID    string
+	Insecure bool
+}
+
+type mmLogger struct {
+	zerolog.Logger
+}
+
+func (l *mmLogger) Output(_ int, s string) error {
+	l.Logger.Info().Msg(s)
+	return nil
+}
+
+// validate that Client adheres to remote interface.
+var _ remote.Remote = (*Client)(nil)
+
+// instantiate a new mattermost client.
+func (c *Client) new() *model.Client4 {
+	log.Info().Msgf("%#v", c)
+
+	url := "https://" + c.Server
+	if c.Insecure {
+		url = "http://" + c.Server
+	}
+
+	log.Info().Msgf("connecting to instance with url: %s", url)
+	api := model.NewAPIv4Client(url)
+	api.SetToken(c.Token)
+
+	return api
+}
+
+func (c *Client) Name() string { return "mattermost" }
+
+func (c *Client) Reaction(_ models.Message, rule models.Rule, _ *models.Bot) {
+	if rule.RemoveReaction != "" {
+		log.Debug().Msg("remove reaction not implemented for mattermost")
+	}
+
+	if rule.Reaction != "" {
+		log.Debug().Msg("reactions not implemented for mattermost")
+	}
+
+}
+
+func (c *Client) Read(inputMsgs chan<- models.Message, _ map[string]models.Rule, bot *models.Bot) {
+	api := c.new()
+	if user, resp, err := api.GetUser("me", ""); err != nil {
+		log.Fatal().Msgf("could not login, %s", err)
+	}
```

Comment:
```suggestion

	user, resp, err := api.GetUser("me", "")
	if err != nil {
		log.Fatal().Msgf("could not login, %s", err)
	}
```

you might have seen from the run, we want to change this up a bit due to:

![image](https://github.com/user-attachments/assets/03c8edd3-417f-43a4-87a5-b74cfa22fa14)

Reply:
Updated. Also fixed some other style issues reported by the linters